code/modules/projectiles/gun.dm

```diff
@@ -212,6 +211,10 @@
 			firing_burst = FALSE
 			return FALSE
 	if(chambered && chambered.BB)
+		if(user.has_trait(TRAIT_PACIFISM))
+			if(chambered.harmful)
+				to_chat(user, "<span class='notice'> [src] is lethal! You don't want to risk harming anyone...</span>")
```

Comment:
The gun doesn't matter, the bullet inside it does. The notification should explain this.